### PR TITLE
allow saving chart images

### LINKED ISSUE #4701: export charts as image
Hi there. One of the most common use cases is to include charts in documents or emails. I was very surprised to see metabase does not have this capability. Maybe I'm not seeing this button?

If someone invests time into getting a pretty good chart/graph then it's because users will find it useful enough to actually use it. Downloading json or csv will force every user to recreate the chart over and over again in the organization.

 I couldn't find any other ticket about this so created this one.

:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

### PR DESCRIPTION
**Pull request by alxnddr**
Closes https://github.com/metabase/metabase/issues/4701

### Description

A last-minute change to allow users to save png images of charts

### How to verify

1. Create a line/area/bar/pie/gauge/... chart and ensure you can save it
2. Add the charts on a dashboard and ensure you can save images from dashcards

Saving images does not work on visualizations that have `canSavePng=false` such as tables, etc.

#### QB:
Screen Shot 2023-02-22 at 3 17 42 PM

#### Dashcards:
Screen Shot 2023-02-22 at 3 18 22 PM

### Checklist

- [ ] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

**Review by kdoh — APPROVED** [review]
Behavior works as expected when I was testing it locally. 👍

File: frontend/src/metabase/icon_paths.ts [hunk 1]
```diff
@@ -143,6 +143,8 @@ export const ICON_PATHS: Record<string, any> = {
   curved:
     "M3.314 25.007c-.398.852-1.427 1.228-2.298.84a1.68 1.68 0 0 1-.86-2.247c3.754-8.047 7.654-12.229 12.06-12.229 2.93 0 4.406 1.185 6.481 4.098l.098.137c1.413 1.984 2.054 2.507 3.318 2.507 2.293 0 4.562-2.814 6.495-8.918.283-.895 1.254-1.396 2.17-1.119.915.277 1.427 1.227 1.144 2.122-2.337 7.38-5.503 11.307-9.809 11.307-2.765 0-4.15-1.132-6.166-3.961l-.097-.137c-1.479-2.075-2.187-2.644-3.635-2.644-2.58 0-5.667 3.31-8.901 10.244z",
   csv: "M28 10.105v18.728A3.166 3.166 0 0 1 24.834 32H6.166A3.163 3.163 0 0 1 3 28.844V3.156A3.163 3.163 0 0 1 6.16 0h13.553V10.105H28zm-.215-1.684h-6.4V.311l6.4 8.11zM17 13v2h2v-2h-2zm0 4v2h2v-2h-2zm4-4v2h2v-2h-2zM7 13v2h7v-2H7zm14 4v2h2v-2h-2zM7 17v2h7v-2H7zm10 4v2h2v-2h-2zm4 0v2h2v-2h-2zM7 21v2h7v-2H7z",
+  // FIXME: update PNG icon
+  png: "M28 10.105v18.728A3.166 3.166 0 0 1 24.834 32H6.166A3.163 3.163 0 0 1 3 28.844V3.156A3.163 3.163 0 0 1 6.16 0h13.553V10.105H28zm-.215-1.684h-6.4V.311l6.4 8.11zM17 13v2h2v-2h-2zm0 4v2h2v-2h-2zm4-4v2h2v-2h-2zM7 13v2h7v-2H7zm14 4v2h2v-2h-2zM7 17v2h7v-2H7zm10 4v2h2v-2h-2zm4 0v2h2v-2h-2zM7 21v2h7v-2H7z",
```
Comment (iethree) [inline frontend/src/metabase/icon_paths.ts:147]: @kdoh were you working on an icon for this?

**Comment by iethree** [comment]
saving from dashboards works great, but I'm getting 0 byte downloads for downloads directly from questions. the console output leads me to believe it thinks the height is 0px?

Screen Shot 2023-02-22 at 5 00 39 PM

**Review by npfitz — APPROVED** [review]
I did notice in the network tab that html2canvas does generate a small 1px gif, maybe that's what Ryan is referring to. I don't think it's an issue though. 

From a UX standpoint, I do find it slightly odd that it gives you *exactly* whats on the screen. When downloading the PNG from the dashboard, if your chart is on a really tiny square, then you download a really tiny image. ex: 

Bar Chart-2_22_2023, 10_27_31 PM

That's probably what we expect, but I was a little surprised when playing with it. On the plus side though, it captures all your series settings!

**Comment by npfitz** [comment]
Looking closer at Ryans screenshot, the 1px gif is not what he is referring to haha. Whoops.

File: frontend/src/metabase/icon_paths.ts (hunk 1 above)
Comment (kdoh) [reply frontend/src/metabase/icon_paths.ts:147]: Yes but it's not blocking to get this out.

**Comment by iethree** [comment]
> saving from dashboards works great, but I'm getting 0 byte downloads for downloads directly from questions. the console output leads me to believe it thinks the height is 0px?

just to document, somehow my loom extension was manipulating the DOM in my browser such that it thought that the chart had zero height. It looks like sasha adjusted the css to force it to full height to fix this.

**Pull request merged**